src/test/java/it/gov/pagopa/payhub/auth/service/a2a/ValidateClientCredentialsServiceTest.java

```diff
@@ -12,27 +15,51 @@
 @ExtendWith(MockitoExtension.class)
 class ValidateClientCredentialsServiceTest {
 
-	@InjectMocks
+	@Mock
+	private ClientService clientService;
 	private ValidateClientCredentialsService service;
 
+	private static final String ALLOWED_CLIENT_ID = "CLIENTID";
 	private static final String ALLOWED_CLIENT_SECRET = "CLIENTSECRET";
 
+	@BeforeEach
+	void setup(){
+		service = new ValidateClientCredentialsService(clientService);
+	}
+
 	@Test
 	void givenValidRequestThenOk() {
+		Mockito.doReturn(new ClientDTO()).when(clientService).verifyCredentials(ALLOWED_CLIENT_ID, ALLOWED_CLIENT_SECRET);
 		assertDoesNotThrow(() ->
-			service.validate(ValidateClientCredentialsService.ALLOWED_SCOPE, ALLOWED_CLIENT_SECRET));
+			service.validate(ALLOWED_CLIENT_ID, ValidateClientCredentialsService.ALLOWED_SCOPE, ALLOWED_CLIENT_SECRET));
+	}
+
+	@Test
+	void givenInvalidClientIdThenInvalidExchangeRequestException() {
+		Mockito.doThrow(new InvalidExchangeRequestException("error"))
+			.when(clientService).verifyCredentials("UNEXPECTED_CLIENT_ID", ALLOWED_CLIENT_SECRET);
+		assertThrows(InvalidExchangeRequestException.class, () ->
+			service.validate("UNEXPECTED_CLIENT_ID", ValidateClientCredentialsService.ALLOWED_SCOPE, ALLOWED_CLIENT_SECRET));
```

Comment:
This is not relevant
InvalidExchangeRequestException is an inner detail of the ClientService, not known outside it.
This exception has not a special management inside the ValidateClientCredentialsService, thus it is not required to test it (otherwise you could think to have to check any possible RuntimeException that could thrown inside any nested invoked method)